.github/workflows/build.yml

```diff
@@ -35,3 +36,45 @@ jobs:
       - name: Run check
         run: |
           SETUPTOOLS_SCM_PRETEND_VERSION="0" make check
+  build-meson:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+          os: [ "alpine", "archlinux", "fedora", "ubuntu" ]
+          include:
+            - install_git_cmd: apk add git
+              os: alpine
+            - install_git_cmd: pacman -Sy --noconfirm git
+              os: archlinux
+            - install_git_cmd: dnf --assumeyes install git
+              os: fedora
+            - install_git_cmd: |
+                export DEBIAN_FRONTEND=noninteractive
+                apt-get update
+                apt-get --assume-yes install git
+              os: ubuntu
+
+    container:
+      image: ${{ matrix.os }}
+
+    steps:
+      # https://github.com/actions/checkout/issues/335
+      - name: Install git
+        run: ${{ matrix.install_git_cmd }}
+
+      - name: Check out source
+        uses: actions/checkout@v3
+
+      - name: Install dependencies
+        run: "${GITHUB_WORKSPACE}/scripts/install-deps.sh"
+
+      # https://github.com/actions/checkout/issues/766
+      - name: Add workspace to safe.directory
+        run: git config --global --add safe.directory "$GITHUB_WORKSPACE"
+
+      - name: Build
+        run: |
+          meson setup . _build
```

Comment:
Once Meson deprecations are fixed in https://github.com/dgibson/dtc/pull/113, we can use `meson setup --fatal-meson-warnings` to enforce version warnings.